modules/ROOT/pages/dataweave-troubleshoot.adoc

```diff
@@ -0,0 +1,105 @@
+= Troubleshooting
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: studio, anypoint, esb, transform, transformer, format, aggregate, rename, split, filter convert, xml, json, csv, pojo, java object, metadata, dataweave, data weave, datamapper, dwl, dfl, dw, output structure, input structure, map, mapping
+
+== Troubleshoot a Failing Script
+
+One challenge to reproducing an error is having the same inputs the script had
+when it executed, especially in production environments where inputs can
+change unexpectedly.
+
+In Mule 4.2.1, DataWeave introduced an experimental feature that enables you
+to dump the input context and the failing script into a folder so that you
+can track the failing script along with the data that makes the script fail.
+
+To use this feature:
+
+. Set the following system property to enable the dump feature: `-M-Dcom.mulesoft.dw.dump_files=true`.
+. [Optional] Specify the path in which to generate the dump files: `-M-Dcom.mulesoft.dw.dump_folder=<path_to_folder>`
++
+The default directory is `/tmp`.
+
+
+== Exception: org.mule.weave.v2.exception.UnsupportedTypeCoercionException
+
+When a function is called with the incorrect kind of argument it throws the
+exception, `org.mule.weave.v2.exception.UnsupportedTypeCoercionException`.
+Causes of this exception include:
+
+* <<null_arg_not_accepted>>
+* <<mimetype_unset>>
+
+[[null_arg_not_accepted]]
+=== Function Does Not Accept Null Argument
+
+The most common cause of this exception occurs when one of the arguments is
+`Null` and the function does not accept `Null` as an argument. This issue
+results in an error message similar to the following:
+
+[source,weave,linenums]
+----
+You called the function '++' with these arguments:
+  1: Null (null)
+  2: String (" A text")
+
+But it expects one of these combinations:
+  (Array, Array)
+  (Date, Time)
+  (Date, LocalTime)
+  (Date, TimeZone)
+  (LocalDateTime, TimeZone)
+  (LocalTime, Date)
+  (LocalTime, TimeZone)
+  (Object, Object)
+  (String, String)
+  (Time, Date)
+  (TimeZone, LocalDateTime)
+  (TimeZone, Date)
+  (TimeZone, LocalTime)
+
+4| payload.message ++ " A text"
+   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Trace:
+  at main::++ (line: 4, column: 1)
+  at main::main (line: 4, column: 17)
+----
+
+One way to resolve this issue uses the `default` operator. For example,
+using `payload.message default "" ++ " A text"` appends empty text when the
+when the message is null.
+
+[[mimetype_unset]]
+=== MIME Type Is Not Set
+
+When the MIME type is not set, you receive an error message similar to the
+following:
+
+[source,weave, linenums]
+----
+You called the function 'Value Selector' with these arguments:
+  1: String ("{message: 123}")
+  2: Name ("message")
+
+But it expects one of these combinations:
```

Comment:
it = the function?

Reply:
"it" is part of the console output. Docs doesn't control that.